### PR TITLE
[ENH] select columns by level

### LINKED ISSUE #1105: Suggestions for `select_columns`
- [x] Allow selection on a level (for MultiIndex columns)
- [x] Return pd.Index if possible (`loc` can handle column selection as an Index)
- [x] Extend slicing to MultiIndex columns via tuples (currently handles only string/integers)
- [x] Allow selection to be generic to hashable, not just strings/tuples (columns can be numeric dtypes)
- [x] As much as possible, pass column presence/absence check to `.loc` (which raises a KeyError if label is not found)

### PR DESCRIPTION
**Pull request by samukweku**
# PR Description

Please describe the changes proposed in the pull request:

- allow column selection by level
- allow slicing on non unique labels that are sorted

**This PR resolves #1105 .**

# PR Checklist

Please ensure that you have done the following:

1. [ ] PR in from a fork off your branch. Do not PR from `<your_username>`:`dev`, but rather from `<your_username>`:`<feature-branch_name>`.

2. [x] If you're not on the contributors list, add yourself to `AUTHORS.md`.

3. [x] Add a line to `CHANGELOG.md` under the latest version header (i.e. the one that is "on deck") describing the contribution.
    - Do use some discretion here; if there are multiple PRs that are related, keep them in a single line.

# Automatic checks

There will be automatic checks run on the PR. These include:

- Building a preview of the docs on Netlify
- Automatically linting the code
- Making sure the code is documented
- Making sure that all tests are passed
- Making sure that code coverage doesn't go down.

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

**Comment by ericmjl** [comment]
🚀 Deployed on https://deploy-preview-1111--pyjanitor.netlify.app

File: janitor/functions/select_columns.py
Comment (ericmjl) [inline janitor/functions/select_columns.py]: @samukweku can I check that `df.columns` as assigned on line 71 is correct? It seems like you assign `df.columns` to something in Line 71 and then reassign it to something else in line 74.

Comment (samukweku) [reply janitor/functions/select_columns.py]: the original columns is saved as `df_columns` at the start. theassignment on line 71 reduces `df.columns` to a single column (so selection can be done). Once the matching labels are obtained, then the original columns are reassigned back in line 74, before the final dataframe is returned.

Comment (ericmjl) [reply janitor/functions/select_columns.py]: Got it! Yes, the explanation makes sense. Before we merge, I think it's important to have the logic documented here. Would you be kind enough add in a few comments in places that you judge would help another maintainer understand the logic?

I'll mark this as the only "change requested" item. Once we're done with this one, I'm good to approve. We'll just need one other approval and then we can merge.

**Review by ericmjl — CHANGES_REQUESTED** [review]
As mentioned in my comment, just one change request to document the code that I was confused about.

**Comment by samukweku** [comment]
Added comments to the updated code. Kindly have a review when U can @ericmjl

**Review by ericmjl — APPROVED** [review]

**Comment by ericmjl** [comment]
Approved! Thank you, @samukweku!

**Comment by ericmjl** [comment]
The PR has been open for about a week. I'd vote that we wait another day before merging.

**Comment by samukweku** [comment]
Thanks for d review @ericmjl , as it prompted me to review d logic

**Comment by ericmjl** [comment]
@samukweku are we good to merge?

**Comment by samukweku** [comment]
@ericmjl kindly give it a day or two ...

**Comment by samukweku** [comment]
@ericmjl ok to merge :grin:

**Comment by ericmjl** [comment]
Merged!! Thank you, @samukweku!

**Pull request merged**
